lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -584,6 +590,74 @@ defmodule Mix.Tasks.Xref do
 
   ## Graph
 
+  defp merge_groups(file_references, pipe_separated_groups) do
+    for group_paths <- pipe_separated_groups,
+        reduce: {file_references, %{}} do
+      {file_references, aliases} ->
+        group_paths
+        |> String.split("|")
```

Comment:
Gah, I just realized that `|` is the pipe operator on the shell, so we will need quoting... or we need to pick something else.

Reply:
I had another idea... what if we allow this to be specified in the `mix.exs` instead? We already have a `xref` key, we could do this:

    xref: [
      groups: [
        "lib-use-modules": ~w(lib/foo.ex lib/foo/web.ex)
      ]
    ]

Reply:
I'd use commas as delimiter (even if technically one could have them in their paths)

Reply:
> I had another idea... what if we allow this to be specified in the `mix.exs` instead? 

So... this would apply to every run of `mix xref graph`??

Reply:
Yes. WDYT?

Reply:
Maybe in the future we could have `--ungroup lib-use-modules` but if we have to ungroup... then perhaps that's not the best idea.

Reply:
In any case, if we are sticking with `--group`, agreed on using `,` as delimiter.

Reply:
> Yes. WDYT?

It doesn't seem intuitive nor really workable to me. In particular, if there is an unexpected dependency from the group, then one will want to inspect the graph with `--lib/foo.ex` but that won't be available anymore.

Reply:
Time for bed, maybe I'll wake up with a different opinion 😸

Reply:
Edited with commas